### DISCUSSION TITLE
Is there any plan for use pnpm?

### DISCUSSION THREAD

**Discussion by aswind7** [fabricjs/fabric.js#10318] [Ideas]
https://pnpm.io/

**Comment by Smrtnyk** [comment]
What is preventing you to use pnpm?

**Comment by asturur** [comment]
Does it require to publish the package in another repository of packages or i just a different flavor of npm like yarn?

**Reply by aswind7** [reply]
Don't need to open a new repository separately, just need to iterate on the existing repository.
In my opinion, npm, yarn and pnpm are three ways of package management, and it can be said that everyone's hobbies are different. It's just that PNPM is a bit faster (it makes use of the soft link cache on the local hard disk), and then it solves the ghost dependency bug in NPM, which is currently used by many open source projects (like Vite and astro).

**Reply by asturur** [reply]
but today if you do pnpm fabric, can you install? what is needed?

**Reply by Smrtnyk** [reply]
to be honest I don't get the issue as well
I am using pnpm in my repo for long time, and using fabric as dependency and never had any issues.
Trying also to understand what fabric has to do with pnpm

**Reply by aswind7** [reply]
Sorry, I may not have explained it clearly. My question is that the open source project fabric currently uses npm to manage dependencies. I want to ask whether this project will use pnpm.

**Reply by asturur** [reply]
No, there are many things to do and that is not the focus.
The basic setup of npm for a library i think is fine, considering that fabricJS has just dev dependencies basically for the command line and testing tools

**Reply by aswind7** [reply]
got it
